circus-train-s3-s3-copier/src/main/java/com/hotels/bdp/circustrain/s3s3copier/aws/JceksAmazonS3ClientFactory.java

```diff
@@ -38,10 +40,17 @@
   private final static Logger LOG = LoggerFactory.getLogger(JceksAmazonS3ClientFactory.class);
 
   private final Security security;
+  private final HiveConf conf;
 
   @Autowired
+  public JceksAmazonS3ClientFactory(Security security, HiveConf replicaHiveConf) {
+    this.security = security;
+    this.conf = replicaHiveConf;
+  }
+
   public JceksAmazonS3ClientFactory(Security security) {
     this.security = security;
+    this.conf = new HiveConf();
```

Comment:
There can be side effects of calling new on a HiveConf so it would be probably be better if we just set/left it null if not set (if we really have to have it).